src/foam/u2/detail/SectionView.js

```diff
@@ -38,37 +55,71 @@ foam.CLASS({
       class: 'Boolean',
       name: 'showTitle',
       value: true
+    },
+    {
+      class: 'FObjectProperty',
+      of: 'foam.core.ArraySlot',
+      name: 'visibilitySlots_',
+      documentation: `
+        An array of slots used to control the visibility of the child
+        SectionedDetailPropertyViews of this view. Used internally to figure out
+        which children are the first and last ones visible so we can adjust the
+        padding on those elements to make sure nested
+        SectionedDetailPropertyViews don't cause unnecessary padding.
+      `,
+      factory: function() {
+        return this.ArraySlot.create();
+      }
+    },
+    {
+      class: 'Int',
+      name: 'firstVisibleIndex_'
+    },
+    {
+      class: 'Int',
+      name: 'lastVisibleIndex_'
     }
   ],
   methods: [
     function initE() {
       var self = this;
-      this.addClass(this.myClass());
       self.SUPER();
-      
+      var proxySlot = self.ProxySlot.create({ delegate$: self.visibilitySlots_$ });
+      self.onDetach(self.firstVisibleIndex_$.follow(proxySlot.map((arr) => arr.indexOf(true))));
+      self.onDetach(self.lastVisibleIndex_$.follow(proxySlot.map((arr) => arr.lastIndexOf(true))));
+
       self
+        .addClass(self.myClass())
         .add(self.slot(function(section, showTitle, section$title) {
           if ( ! section ) return;
-          return self.Rows.create()
+          var slots = [];
+          var elm = self.Rows.create()
             .show(section.createIsAvailableFor(self.data$))
-            .callIf(showTitle && section$title, function () {
+            .callIf(showTitle && section$title, function() {
               this.start('h2').add(section$title).end();
             })
             .start(self.Grid)
-              .forEach(section.properties, function (p) {
-                this.start(self.GUnit, { 
-                  columns: p.gridColumns 
-                })
-                  .tag(self.SectionedDetailPropertyView, { prop: p, data$: self.data$ })
-                .end()
+              .forEach(section.properties, function(p, index) {
+                this.start(self.GUnit, { columns: p.gridColumns })
+                  .start(self.SectionedDetailPropertyView, {
+                    prop: p,
+                    data$: self.data$
+                  })
+                    .call(function() { slots.push(this.visibilitySlot); })
+                    .enableClass('first', self.firstVisibleIndex_$.map((value) => value === index))
+                    .enableClass('last', self.lastVisibleIndex_$.map((value) => value === index))
+                  .end()
+                .end();
               })
             .end()
             .start(self.Cols)
-              .forEach(section.actions, function (a) {
+              .forEach(section.actions, function(a) {
                 this.add(a);
               })
             .end();
+          this.visibilitySlots_ = self.ArraySlot.create({ slots: slots });
```

Comment:
this becomes `proxySlot.delegate = self.ArraySlot.create({ slots: slots });`

Reply:
Resolved by fd57cd2.